quadriga/src/main/java/edu/asu/diging/quadriga/web/EditCollectionController.java

```diff
@@ -0,0 +1,88 @@
+package edu.asu.diging.quadriga.web;
+
+import java.util.Objects;
+
+import javax.validation.Valid;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import edu.asu.diging.quadriga.core.exceptions.CollectionNotFoundException;
+import edu.asu.diging.quadriga.core.model.Collection;
+import edu.asu.diging.quadriga.core.service.CollectionManager;
+import edu.asu.diging.quadriga.web.forms.CollectionForm;
+
+/**
+ * This controller enables users to edit their collections
+ * 
+ * @author poojakulkarni
+ *
+ */
+@Controller
+public class EditCollectionController {
+
+    @Autowired
+    private CollectionManager collectionManager;
+
+    /**
+     * Request handler for getting the "Edit collections" view
+     * 
+     * @param id    of the collection to be edited
+     * @param model to set collectionForm attribute to pre-fill current name and
+     *              description
+     * @return the editCollection view
+     */
+    @RequestMapping(value = "/auth/collections/edit/{id}", method = RequestMethod.GET)
```

Comment:
path should be `/auth/collections/{id}/edit` so the base path is the same for all links related to a collection and just the suffix changes (e.g. `/auth/collections/{id}/delete`, `/auth/collections/{id}`)